app_dart/lib/src/request_handlers/override_github_build_status.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/service/github_service.dart';
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../foundation/providers.dart';
+import '../foundation/typedefs.dart';
+import '../model/appengine/github_build_status_update.dart';
+import '../model/appengine/github_tree_status_override.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../service/datastore.dart';
+
+@immutable
+class OverrideGitHubBuildStatus extends ApiRequestHandler<Body> {
+  const OverrideGitHubBuildStatus(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+    @visibleForTesting LoggingProvider loggingProvider,
+  })  : datastoreProvider = datastoreProvider ?? DatastoreService.defaultProvider,
+        loggingProvider = loggingProvider ?? Providers.serviceScopeLogger,
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+  final LoggingProvider loggingProvider;
+
+  static const String repositoryKeyName = 'repository';
+  static const String closedKeyName = 'closed';
+  static const String reasonKeyName = 'reason';
+  static const String forceKeyName = 'force';
```

Comment:
would it be possible to use an enum instead?

Reply:
These are keys for a rest request.

I guess I could instead create a serialized type though. That would probably be a bit easier to reason about.